topdown/crypto_test.go

```diff
@@ -124,3 +124,43 @@ func TestCryptoSha256(t *testing.T) {
 	}
 
 }
+
+func TestCryptoX509ParseCertificateRequest(t *testing.T) {
+	rule := `
+		p[x] {
+			parsed := crypto.x509.parse_certificate_request(cert)
+			x := parsed.Subject.CommonName
+		}
+	`
+
+	tests := []struct {
+		note     string
+		cert     string
+		rule     string
+		expected interface{}
+	}{
+		{
+			note:     "one",
+			cert:     `LS0tLS1CRUdJTiBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0KTUlJQllUQ0NBUWdDQVFBd01ERXVNQ3dHQTFVRUF4TWxiWGt0Y0c5a0xtMTVMVzVoYldWemNHRmpaUzV3YjJRdQpZMngxYzNSbGNpNXNiMk5oYkRCWk1CTUdCeXFHU000OUFnRUdDQ3FHU000OUF3RUhBMElBQk1DYzN6eE9TYkN2Ck9NYitoajh2MW9aU2tPdHhLMXphdmVXck9ZVVJ5WTJOUU8wZFBmN2JuRkNZSEhKRFBwMUJTSDFwK2xXMS83MjYKdU5qTUNFdnRPdGlnZGpCMEJna3Foa2lHOXcwQkNRNHhaekJsTUdNR0ExVWRFUVJjTUZxQ0pXMTVMWE4yWXk1dAplUzF1WVcxbGMzQmhZMlV1YzNaakxtTnNkWE4wWlhJdWJHOWpZV3lDSlcxNUxYQnZaQzV0ZVMxdVlXMWxjM0JoClkyVXVjRzlrTG1Oc2RYTjBaWEl1Ykc5allXeUhCTUFBQWhpSEJBb0FJZ0l3Q2dZSUtvWkl6ajBFQXdJRFJ3QXcKUkFJZ1o5RXNFNTlaZG9PWSs4Mm9Cc1Q1bUd3a2p6WDBqdFJqci9OazIzVVBqcUlDSUVXVk56T2wzSCtqZTA2MwpWRXhTQ080ZzBUSHNiTGhidXVFc1NCYS9VUVBXCi0tLS0tRU5EIENFUlRJRklDQVRFIFJFUVVFU1QtLS0tLQ==`,
+			rule:     rule,
+			expected: `["my-pod.my-namespace.pod.cluster.local"]`,
+		},
+		{
+			note:     "bad",
+			cert:     `LS0tLS1CRUdJTiBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0=`,
+			rule:     rule,
+			expected: &Error{Code: BuiltinErr, Message: "invalid PEM-encoded certificate signing request"},
+		},
```

Comment:
Lets add a couple more negative tests:

* Passing in a non-string value
* Passing in a non base64 encoded (but otherwise valid) request

nit: Please add a tiny bit more descriptive "note" for each too, these ultimately end up being the test name so a little more detail about the test case is often useful

Reply:
Done. better notes too.